crates/proof-of-sql/src/sql/ast/numerical_util.rs

```diff
@@ -82,6 +121,23 @@ pub(crate) fn add_subtract_columns<'a, S: Scalar>(
     res
 }
 
+/// Multiply two columns together.
+pub(crate) fn multiply_columns<'a, S: Scalar>(
+    lhs: &Column<'a, S>,
+    rhs: &Column<'a, S>,
+    alloc: &'a Bump,
+) -> &'a [S] {
+    let lhs_len = lhs.len();
+    let rhs_len = rhs.len();
+    assert!(
+        lhs_len == rhs_len,
+        "lhs and rhs should have the same length"
+    );
+    alloc.alloc_slice_fill_with(lhs_len, |i| {
+        lhs.scalar_at(i).unwrap() * rhs.scalar_at(i).unwrap()
```

Comment:
NOTE: This is fine for now, but I think we need to revise things like this eventually.
I don't like blanket converting to scalars here. Aside from type safety, which is perhaps unavoidable, this is an efficiency concern.
For example:
```
let a = i64::MIN
let b = i64::MAX
let prod = S::from(a) * S::from(b)
```
is less efficient than
```
let a = i64::MIN
let b = i64::MAX
let prod = S::from((a as i128) * (b as i128))
```

This certainly would lead to larger `match` branches, but in general, I like the philosophy of "only convert to Scalar when necessary".

This is a very minor performance change in this case, so we can do the optimization later, but it is something that might eventually give very real performance gains with some of the optimizations I have in mind.